### PR TITLE
Add Image Display plugin

### PR DESCRIPTION
**Pull request by Sarath18**
I have also written two scripts to publish images in `bgr8`, `rgb8` and `mono8` format from webcam and video file. 
This can be found [here](https://gist.github.com/Sarath18/7c48b6f2e667bf6dab1b12f419cab397).

### CODE REVIEW

File: ign_rviz_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -89,6 +89,33 @@ target_include_directories(GlobalOptions
...
+########################################################################
+add_library(ImageDisplay SHARED ${headers_MOC}
+  include/ignition/rviz/plugins/ImageDisplay.hpp
+  include/ignition/rviz/plugins/message_display_base.hpp
+  src/rviz/plugins/ImageDisplay.cpp
+  ${resources_RCC}
+)
+
+ament_target_dependencies(ImageDisplay
+  tf2_ros
```
Comment (ahcorde) [inline ign_rviz_plugins/CMakeLists.txt:101]: alphabetize

File: ign_rviz_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -89,6 +89,33 @@ target_include_directories(GlobalOptions
...
+  sensor_msgs
+  ignition-gui4
+  ignition-math6
+  ignition-rendering4
+  ign_rviz_common
```
Comment (ahcorde) [inline ign_rviz_plugins/CMakeLists.txt:106]:
```suggestion
ign_rviz_common
ignition-gui4
ignition-math6
ignition-rendering4
sensor_msgs
tf2_ros
```

File: ign_rviz_plugins/include/ignition/rviz/plugins/ImageDisplay.hpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+   * @param index Combo box index
+   */
+  void setCurrentIndex(const int index);
+
+  /**
+   * @brief  Notify that a new image has been received
+   */
+
+signals:
+  void newImage();
```
Comment (ahcorde) [inline ign_rviz_plugins/include/ignition/rviz/plugins/ImageDisplay.hpp:143]:
```suggestion
signals:
   /**
    * @brief  Notify that a new image has been received
    */
    void newImage();
```

File: ign_rviz_plugins/res/qml/ImageDisplay.qml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    anchors.right: parent.right
+
+    anchors.topMargin: 2
+    anchors.leftMargin: -5
+    anchors.rightMargin: -5
+
+    Layout.fillHeight: true
+    Layout.fillWidth: true
+    // width: parent.width
+    // height: parent.height
```
Comment (ahcorde) [inline ign_rviz_plugins/res/qml/ImageDisplay.qml:105]: can we remove this two lines ?

**Comment by ahcorde** [comment]
Do you mind to add some instructions?

I used 
```
ros2 launch ros_ign_gazebo_demos camera.launch.py
```

**Comment by Sarath18** [comment]
Sure no issue. I wanted to ask if the instruction should be from `ros_ign_gazebo_demos`? 
I think it might be possible to use we can use ros2 demos instead
```bash
ros2 run image_tools cam2image
```
Also, where should I add this instruction? README or start documenting plugins in the wiki?

**Comment by ahcorde** [comment]
Add a section in the README.md. I was not able to run the command that you mention. I have some issues with `gstreamer`. Anyhow I don't mind which one you choose

**Pull request merged**